vllm/sequence.py

```diff
@@ -469,41 +432,48 @@ def n_blocks(self) -> int:
 
     @cached_property
     def prompt(self) -> Optional[str]:
-        # Select decoder or encoder input prompt str, as appropriate
-        prompt_key: str = ("prompt"
-                           if self.from_decoder_prompt else "encoder_prompt")
+        inputs = self.inputs
+
+        if inputs["type"] == "token":
+            return inputs.get("prompt")
 
-        return cast(Optional[str], self.inputs.get(prompt_key))
+        assert_never(inputs)
```

Comment:
Is there a reason to not write these like this? (which is clearer imo)
```python
    assert inputs["type"] == "token"
    return inputs.get("prompt")
```

Reply:
It's to reduce diffs when introducing #6869 which adds a new `inputs["type"] == "embed"` branch.